paasta_tools/cli/cmds/validate.py

```diff
@@ -572,7 +598,96 @@ def __is_templated(service: str, soa_dir: str, cluster: str, workload: str) -> b
     )
 
 
-def validate_autoscaling_configs(service_path):
+def _validate_active_requests_autoscaling_configs(
+    instance_config: InstanceConfig,
+    autoscaling_params: AutoscalingParamsDict,
+) -> None:
+    desired_active_requests_per_replica = autoscaling_params.get(
+        "desired_active_requests_per_replica",
+        DEFAULT_DESIRED_ACTIVE_REQUESTS_PER_REPLICA,
+    )
+    if desired_active_requests_per_replica <= 0:
+        raise AutoscalingValidationError(
+            "desired_active_requests_per_replica must be positive"
+        )
+    if len(cast(LongRunningServiceConfig, instance_config).get_registrations()) > 1:
+        raise AutoscalingValidationError(
+            "active-requests metrics provider doesn't support instances with multiple registrations"
+        )
+    # This is a bit incorrect, since the default autoscaling setpoint is 0.8,
+    # someone could theoretically bypass this by setting setpoint: 0.8 in their
+    # soaconfigs; but I think it's approximately fine for now
+    if (
+        autoscaling_params.get("setpoint", DEFAULT_AUTOSCALING_SETPOINT)
```

Comment:
i was just thinking that it might be more future-proof to have an allowlist of `$provider: $valid_params` so that we don't have all these methods more or less checking similar fields - but that sounds like something we can refactor later :)

Reply:
Oh that's probably a good idea.

Reply:
Actually - I'm going to do that in #3837